frontend/src/components/headers/DocumentHeader.tsx

```diff
@@ -46,6 +65,36 @@ function DocumentHeader() {
 		navigate(`/${workspaceState.data?.slug}`);
 	};
 
+	const handleDocumentTitleChange = (
+		e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
+	) => {
+		if (documentStore.data) {
+			dispatch(
+				setDocumentData({
+					...documentStore.data,
+					title: e.target.value,
+				})
+			);
+		}
+	};
```

Comment:
We don't have to update the document in store.
It is automatically updated by `hooks/api/workspaceDocument.ts`.